x-pack/plugins/security_solution/server/fleet_integration/handlers/remove_policy_from_artifacts.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import pMap from 'p-map';
+
+import {
+  ENDPOINT_TRUSTED_APPS_LIST_ID,
+  ENDPOINT_EVENT_FILTERS_LIST_ID,
+  ENDPOINT_HOST_ISOLATION_EXCEPTIONS_LIST_ID,
+} from '@kbn/securitysolution-list-constants';
+import { ExceptionListClient } from '../../../../lists/server';
+import { PostPackagePolicyDeleteCallback } from '../../../../fleet/server';
+
+export const ARTIFACT_LISTS_IDS_TO_REMOVE = [
+  ENDPOINT_TRUSTED_APPS_LIST_ID,
+  ENDPOINT_EVENT_FILTERS_LIST_ID,
+  ENDPOINT_HOST_ISOLATION_EXCEPTIONS_LIST_ID,
+];
+
+/**
+ * Removes policy from artifacts
+ */
+export const removePolicyFromArtifacts = async (
+  exceptionsClient: ExceptionListClient,
+  policy: Parameters<PostPackagePolicyDeleteCallback>[0][0]
+) => {
+  let page = 1;
+
+  const findArtifactsByPolicy = async (currentPage: number) => {
```

Comment:
`async` is redundant when we're returning the promise here and not `await`ing it.